Snowflake.Data/Core/SnowflakeTomlConnectionBuilder.cs

```diff
@@ -0,0 +1,128 @@
+// <copyright file="SnowflakeTomlConnectionBuilder.cs" company="Snowflake Inc">
+//         Copyright (c) 2024 Snowflake Inc. All rights reserved.
+//  </copyright>
+
+namespace Snowflake.Data.Core
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using Client;
+    using Log;
+    using Tomlyn;
+    using Tomlyn.Model;
+    using Tools;
+
+    public class SnowflakeTomlConnectionBuilder
+    {
+        private const string DefaultConnectionName = "default";
+        private const string DefaultSnowflakeFolder = ".snowflake";
+        private const string DefaultTokenPath = "/snowflake/session/token";
+
+        private readonly SFLogger _logger = SFLoggerFactory.GetLogger<SnowflakeDbConnection>();
+
+        private readonly Dictionary<string, string> _tomlToNetPropertiesMapper = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase)
+        {
+            { "DATABASE", "DB" }
+        };
+
+        private readonly FileOperations _fileOperations;
+        private readonly EnvironmentOperations _environmentOperations;
+
+        public SnowflakeTomlConnectionBuilder() : this(FileOperations.Instance, EnvironmentOperations.Instance)
+        {
+        }
+
+        internal SnowflakeTomlConnectionBuilder(FileOperations fileOperations, EnvironmentOperations environmentOperations)
+        {
+            _fileOperations = fileOperations;
+            _environmentOperations = environmentOperations;
+        }
+
+        public string GetConnectionStringFromToml(string connectionName = null)
+        {
+            var connectionString = string.Empty;
+            var tomlPath = ResolveConnectionTomlFile();
+            var connectionToml = GetTomlTableFromConfig(tomlPath, connectionName);
+            if (connectionToml != null)
+            {
+                connectionString = GetConnectionStringFromTomlTable(connectionToml);
+            }
+            return connectionString;
+        }
+
+        private string GetConnectionStringFromTomlTable(TomlTable connectionToml)
+        {
+            var connectionStringBuilder = new StringBuilder();
```

Comment:
To include a semicolon it should be following the same rules as for a normal ConnectrionString, it should be double quoted 

``` toml
[default]
host = "fakeaccount.snowflakecomputing.com"
user = "fakeuser"
password = "\";fakepassword\""
```

Reply:
Updated documentation with samples

Reply:
I added a sample for the quotation when including the case for the semicolon, it would be the same in any position. 
https://github.com/snowflakedb/snowflake-connector-net/blob/b284fc1655869c62f053d122b23c1c56ce05f00d/doc/Connecting.md?plain=1#L351-L358

Reply:
ok, you could also think of creating a test for that. Probably in TomlConnectionBuilderTest.

Reply:
Done, added test for special characters.